x/valset-pref/types/msgs.go

```diff
@@ -49,9 +49,21 @@ func (m MsgSetValidatorSetPreference) ValidateBasic() error {
 		return fmt.Errorf("The validator operator address are duplicated")
 	}
 
+	// Round to 2 digit after the decimal. For ex: 0.999 = 1.0, 0.874 = 0.87, 0.5123 = 0.51
+	totalWeightFloat64, err := totalWeight.Float64()
+	if err != nil {
+		return err
+	}
+
+	roundedValueStr := fmt.Sprintf("%.2f", totalWeightFloat64)
+	roundedValue, err := strconv.ParseFloat(roundedValueStr, 64)
+	if err != nil {
+		return err
+	}
```

Comment:
Why was this check added?

Reply:
currently if the delegator provides {0.3333, 0.6666} as weights it'd output 0.9999 which is not equal to 1, which i think we should round up